x-pack/plugins/ml/public/application/datavisualizer/index_based/components/field_data_row/action_menu/lens_utils.ts

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ML_JOB_FIELD_TYPES } from '../../../../../../../common/constants/field_types';
+import type { TypedLensByValueInput } from '../../../../../../../../lens/public';
+import type { IndexPattern } from '../../../../../../../../../../src/plugins/data/common/index_patterns/index_patterns';
+import type { FieldVisConfig } from '../../../../stats_table/types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { DataType } from '../../../../../../../../lens/public/types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { IndexPatternColumn } from '../../../../../../../../lens/public/indexpattern_datasource/operations/definitions';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { LayerConfig } from '../../../../../../../../lens/public/xy_visualization/types';
+import type { CombinedQuery } from '../../../common';
+
+interface ColumnsAndLayer {
+  columns: Record<string, IndexPatternColumn>;
+  layer: LayerConfig;
+}
+
+export function getNumberSettings(item: FieldVisConfig, defaultIndexPattern: IndexPattern) {
+  // if index has no timestamp field
+  if (defaultIndexPattern.timeFieldName === undefined) {
+    const columns: Record<string, IndexPatternColumn> = {
+      col1: {
+        label: item.fieldName!,
+        dataType: 'number',
+        isBucketed: true,
+        operationType: 'range',
+        params: {
+          type: 'histogram',
+          maxBars: 'auto',
+          ranges: [],
+        },
+        sourceField: item.fieldName!,
+      },
+      col2: {
+        label: 'Count',
```

Comment:
does this string require translation?

Reply:
Updated here https://github.com/elastic/kibana/pull/89471/commits/a84a88526266049f3ded62fc4cbd666627b9132e